### ISSUE TITLE
Route params typefication

### ISSUE THREAD

**Issue by ricardocastanho** [nuxt/nuxt#20299]
### Describe the feature

When using the `useRoute` composable, the route params aren't beeing typed.
I'm in the folder's path `pasges/me/[id].vue` and when I type the code below, the `id` insn't in the vscode completition.

```javascript
<script setup>
const route = useRoute();
console.log(route.params.id) // id isn't in the vscode completition
</script>
```

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by Lehoczky** [comment]
Have you tried out the [nuxt typed router module](https://nuxt-typed-router.vercel.app/)? Seems like it should do what you are looking for

**Comment by danielroe** [comment]
This has now landed with https://github.com/nuxt/nuxt/pull/20367, which is available in Nuxt v3.5.
